ddsrouter_utils/include/ddsrouter_utils/memory/OwnerPtr.hpp

```diff
@@ -0,0 +1,320 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file OwnerPtr.hpp
+ *
+ * This file contains class OwnerPtr implementation.
```

Comment:
I've thought a lot about it. I decided to leave it together because it would not make sense to use one without the other (to include one and not the other) but it could be separated as well